spring-integration-core/src/test/java/org/springframework/integration/config/annotation/AnnotatedEndpointActivationTests.java

```diff
@@ -80,6 +82,7 @@ public void sendAndReceive() {
 
 		assertTrue(this.applicationContext.containsBean("annotatedEndpoint.process.serviceActivator"));
 		assertTrue(this.applicationContext.containsBean("annotatedEndpoint2.process.serviceActivator"));
+		assertTrue(this.applicationContext.containsBean("annotatedEndpoint3.process.serviceActivator"));
```

Comment:
The test isn't full. It doesn't prove that `async` is applied. 
I'm sure the test will pass even without that `async = "true"`.

Please, develop the test-case to reflect the expectations.

Reply:
That's not entirely true; it will fail sometimes since the `ListenableFuture` doesn't get unwrapped. I can separate it out into a separate test case though if you would prefer?